corehq/project_limits/rate_limiter.py

```diff
@@ -176,3 +177,19 @@ def get_rate_limits(self):
             (self.per_minute, minute_rate_counter),
             (self.per_second, second_rate_counter),
         ) if limit]
+
+
+def get_dynamic_rate_definition(key, default):
+    dynamic_rate_definition, _ = DynamicRateDefinition.objects.get_or_create(
+        key=key, defaults=_get_rate_definition_dict(default))
+    return RateDefinition(**_get_rate_definition_dict(dynamic_rate_definition))
+
+
+def _get_rate_definition_dict(rate_definition):
+    """
+    Convert RateDefinition-like object to dict like {'per_week': ..., 'per_hour': ..., ...}
+    """
+    return {
+        attribute.name: getattr(rate_definition, attribute.name)
+        for attribute in RateDefinition.__attrs_attrs__
```

Comment:
What do you think of adding a test to ensure the attributes of `RateDefinition` match those of `DynamicRateDefinition` (possibly with some exclusions)? I'm envisioning such a test to fail if someone ever changed one of them in a way that would break this code or make them generally incompatible.

Reply:
Yup, this general feature was imminently testable, and the tests make it better. Thanks for the reminder to be good about adding tests!

For this specific test I decided to make sure the functionality is tested, but without specific tests for this low-level detail. It's not clear cut to me, but I added the more sort of "interface level" tests, and checked that they also cover this, and at that point it seemed like having this specific wouldn't have a good brittleness / catches new problems ratio.